src/SquareRootEngine.cpp

```diff
@@ -0,0 +1,88 @@
+#include "SquareRootEngine.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <stdexcept>
+
+#include "Size.hpp"
+#include "utility.hpp"
+
+SquareRootEngine::SquareRootEngine(int step_count, int max_speed)
+    : mStepCount(step_count), mMaxSpeed(max_speed) {}
+
+void SquareRootEngine::set_gear(Gear gear) {
+  if (mCurrentGear == Gear::Neutral && gear != Gear::Neutral) {
+    mStep = get_step_for_current_speed();
+  }
+  mCurrentGear = gear;
+}
+
+int SquareRootEngine::get_step_for_current_speed() const {
+  if (mCurrentSpeed == 0) {
+    return 1;
+  }
+  return std::pow(mCurrentSpeed * std::sqrt(mStepCount) / mMaxSpeed, 2);
+}
+
+float SquareRootEngine::get_current_speed() const { return mCurrentSpeed; }
+
+void SquareRootEngine::update() {
+  mBrake = update_brake();
+  mCurrentSpeed = update_current_speed();
+  mStep = update_step();
+}
+
+bool SquareRootEngine::update_brake() const {
+  if (mCurrentGear == Gear::Reverse && mCurrentSpeed > 0) {
+    return true;
+  }
+  if (mCurrentGear == Gear::Drive && mCurrentSpeed < 0) {
+    return true;
+  }
+  return false;
+}
+
+int SquareRootEngine::update_step() const {
```

Comment:
`update` methods suggests that you are updating something. Please rename to `calculate`, `get` or something similar